From 295305c914d7a9e9e49c35e0ca921a64251a443e Mon Sep 17 00:00:00 2001
From: Tobias Kaufmann <Tobias.KA.Kaufmann@bmw.de>
Date: Tue, 13 Dec 2022 14:35:42 +0100
Subject: [PATCH 6/8] CMakelists.txt: Add stack-protector-strong flag

Enable stack canaries to mitigate buffer overflows on the stack.
---
 ref/CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/ref/CMakeLists.txt b/ref/CMakeLists.txt
index 00d960f..72a69d7 100644
--- a/ref/CMakeLists.txt
+++ b/ref/CMakeLists.txt
@@ -24,6 +24,7 @@ else()
   add_compile_options(-Wall -Wextra -Wpedantic -Werror)
   add_compile_options(-Wmissing-prototypes -Wredundant-decls -Wshadow -Wpointer-arith)
   add_compile_options(-O3 -fomit-frame-pointer)
+  add_compile_options(-fstack-protector-strong)
 endif()
 
 add_library(fips202_ref STATIC ${FIPS202_SRCS})
-- 
2.38.1

